Initial commit for OCRSpace connector.

bidisha-spryiq opened the pull request.

bidisha-spryiq committed:
Initial commit for OCRSpace connector.

cs-prerna-joshi commented on OCRSpace/info.json:
```diff
@@ -0,0 +1,176 @@
+    "description": "The OCR API provides a simple way of parsing images and multi-page PDF documents (PDF OCR) and getting the extracted text results returned in a JSON format.",
+    "publisher": "spryIQ.co",
+    "icon_small_name": "ocrspacelogo32x32.png",
+    "icon_large_name": "ocrspacelogo100x100.png",
```
```suggestion
    "icon_large_name": "ocrspacelogo80x80.png",
```

cs-prerna-joshi commented on OCRSpace/info.json:
```diff
@@ -0,0 +1,176 @@
+    "publisher": "spryIQ.co",
+    "icon_small_name": "ocrspacelogo32x32.png",
+    "icon_large_name": "ocrspacelogo100x100.png",
+    "category": "Text Extraction",
```
```suggestion
    "category": "utilities",
```

cs-prerna-joshi commented on OCRSpace/info.json:
```diff
@@ -0,0 +1,176 @@
+            },
+            {
+                "title": "API Key",
+                "type": "text",
```
```suggestion
                "type": "password",
```

cs-prerna-joshi commented on OCRSpace/info.json:
```diff
@@ -0,0 +1,176 @@
+            },
+            {
+                "title": "Language",
+                "type": "text",
```
if there are pre defined set of Languages, then you can give those in options like...

"type": "select",
"options": ["eng", "spanish", "german" ]

It will be easy for user to select from these options.

cs-prerna-joshi commented on OCRSpace/info.json:
```diff
@@ -0,0 +1,176 @@
+                    }
+                ],
+                "OCRExitCode": "",
+                "IsErroredOnProcessing": false,
```
```suggestion
                "IsErroredOnProcessing":"",
```

cs-prerna-joshi commented on OCRSpace/info.json:
```diff
@@ -0,0 +1,176 @@
+                    "onchange": {
+                        "url": [
+                            {
+                                "title": "imageURL",
```
```suggestion
                                "title": "Image URL",
```

cs-prerna-joshi commented on OCRSpace/info.json:
```diff
@@ -0,0 +1,176 @@
+                        ],
+                        "base64image": [
+                            {
+                                "title": "Base64Image",
```
```suggestion
                                "title": "Base64 Image",
```

cs-prerna-joshi commented on OCRSpace/operations.py:
```diff
@@ -0,0 +1,137 @@
+""" Copyright start
+  Copyright (C) 2008 - 2021 Fortinet Inc.
```
```suggestion
  Copyright (C) 2008 - 2022 Fortinet Inc.
```

cs-prerna-joshi commented on OCRSpace/operations.py:
```diff
@@ -0,0 +1,137 @@
+        try:
+            file_path = join('/tmp', download_file_from_cyops(file_iri)['cyops_file_path'])
+            logger.info(file_path)
+            out = subprocess.Popen(["file", str(file_path)], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```
for this part please refer [(https://github.com/fortinet-fortisoar/connector-hybrid-analysis/blob/develop/hybrid-analysis/operations.py)]

bidisha-spryiq replied:
Thanks for providing this reference. I went through it and observe that most of the solution in this code is similar to the given reference except subprocess command I have used in line #57. My requirement is to automatically detect the type of file given by the user and I don't find anything related to this in the referred module.
Can you please specifically point out what you want me to change here?

cs-prerna-joshi commented on OCRSpace/playbooks/playbooks.json:
```diff
@@ -0,0 +1,388 @@
+{
```
please update the PBs using the script.

bidisha-spryiq committed:
1. Addressed review comments.
2. Refactored the folder name and code based on the connector development checklist provided.
